src/Engine/Manager/Managers.cpp

```diff
@@ -55,74 +53,16 @@ void Hub::ShutDown() {
-        soundManager->AddComponent(world, component, componentType);
-    else if (*componentType == typeid(Component::Material*))
-        renderManager->AddComponent(world, component, componentType);
-    else if (*componentType == typeid(Component::Mesh*))
-        renderManager->AddComponent(world, component, componentType);
-    else if (*componentType == typeid(Component::ParticleEmitter*))
-        particleManager->AddComponent(world, component, componentType);
-    else if (*componentType == typeid(Component::Physics*))
-        physicsManager->AddComponent(world, component, componentType);
-    else if (*componentType == typeid(Component::PointLight*))
-        renderManager->AddComponent(world, component, componentType);
-    else if (*componentType == typeid(Component::Script*))
-        scriptManager->AddComponent(world, component, componentType);
-    else if (*componentType == typeid(Component::SoundSource*))
-        soundManager->AddComponent(world, component, componentType);
-    else if (*componentType == typeid(Component::SpotLight*))
-        renderManager->AddComponent(world, component, componentType);
-    else
-        Log() << componentType->name() << " not assigned to a manager!" << "\n";
-}
-
-void Hub::ClearComponents(World* world) {
-    if (!shutdown) {
-        resourceManager->ClearComponents(world);
-        renderManager->ClearComponents(world);
-        particleManager->ClearComponents(world);
-        physicsManager->ClearComponents(world);
-        soundManager->ClearComponents(world);
-        scriptManager->ClearComponents(world);
-        debugDrawingManager->ClearComponents(world);
-        profilingManager->ClearComponents(world);
-    }
-}
-
-void Hub::ClearAllComponents() {
-    if (!shutdown) {
-        resourceManager->ClearAllComponents();
-        renderManager->ClearAllComponents();
-        particleManager->ClearAllComponents();
-        physicsManager->ClearAllComponents();
-        soundManager->ClearAllComponents();
-        scriptManager->ClearAllComponents();
-        debugDrawingManager->ClearAllComponents();
-        profilingManager->ClearAllComponents();
-    }
-}
-
-void Hub::ClearKilledComponents(World* world) {
+void Hub::ClearKilledComponents() {
     if (!shutdown) {
-        resourceManager->ClearKilledComponents(world);
-        renderManager->ClearKilledComponents(world);
-        particleManager->ClearKilledComponents(world);
-        physicsManager->ClearKilledComponents(world);
-        soundManager->ClearKilledComponents(world);
-        scriptManager->ClearKilledComponents(world);
-        debugDrawingManager->ClearKilledComponents(world);
-        profilingManager->ClearKilledComponents(world);
+        renderManager->ClearKilledComponents();
+        particleManager->ClearKilledComponents();
+        physicsManager->ClearKilledComponents();
+        soundManager->ClearKilledComponents();
+        scriptManager->ClearKilledComponents();
```

Comment:
`resourceManager`, `debugDrawingManager`, and `profilingManager` are still members of `Hub`. Is there any particular reason they are not cleared?

Reply:
They don't contain any components

Reply:
Cool.